### ISSUE TITLE
Typescript debug failure, causes in javascript files folding/unfolding functionality disappear (content remains hidden)

### ISSUE THREAD

**Issue by 5saviahv** [microsoft/vscode#112027]
- VSCode Version: 1.51.1
- OS Version: Linux x64 4.15.0-126-generic

Steps to Reproduce:

1. I use old nodejs v10.15.0 source as reference for understanding how streams work.
2. I open "lib/_stream_readable.js"
3. I take vscode suggestion for converting functions to ES6 classes.

First all folding/unfolding work, but all the sudden all ability to fold/unfold disappears. you also can't use file any longer, since folded content remains hidden. 

* I also tried copy this file into another location and simply waited and it happens again.

![Screenshot from 2020-12-07 17-03-34](https://user-images.githubusercontent.com/49443574/101370971-e018fc00-38b2-11eb-8cf7-310195e23277.png)

After it happens:
* I tried close and open vscode, first is seems to work, but later it didn't help any longer.
* Then I made copy from file and when it happens I add some spaces and hit save and folding/unfolding appear again.

<details><summary>"Log (Window)" contains some typescript errors:</summary><p>

```
[2020-12-07 16:33:19.977] [renderer6] [error] <syntax> TypeScript Server Error (4.0.3)
Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
Error: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
    at addSyntheticNodes (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142388:30)
    at createChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142377:9)
    at NodeObject.getChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142321:56)
    at Object.findChildOfKind (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:111269:26)
    at spanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123467:36)
    at getOutliningSpanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123387:28)
    at visitNonImportNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123233:28)
    at addNodeOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123204:21)
    at Object.collectElements (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123191:13)
    at Proxy.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:143739:50)
    at IOSession.Session.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154172:45)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:153172:61)
    at /usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:88
    at IOSession.Session.executeWithRequestId (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154955:28)
    at IOSession.Session.executeCommand (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:33)
    at IOSession.Session.onMessage (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154988:35)
    at Interface.<anonymous> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:157204:27)
    at Interface.emit (events.js:223:5)
    at Interface._onLine (readline.js:315:10)
    at Interface._normalWrite (readline.js:460:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:223:5)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:290:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:181:23): Error: <syntax> TypeScript Server Error (4.0.3)
Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
Error: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
    at addSyntheticNodes (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142388:30)
    at createChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142377:9)
    at NodeObject.getChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142321:56)
    at Object.findChildOfKind (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:111269:26)
    at spanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123467:36)
    at getOutliningSpanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123387:28)
    at visitNonImportNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123233:28)
    at addNodeOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123204:21)
    at Object.collectElements (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123191:13)
    at Proxy.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:143739:50)
    at IOSession.Session.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154172:45)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:153172:61)
    at /usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:88
    at IOSession.Session.executeWithRequestId (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154955:28)
    at IOSession.Session.executeCommand (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:33)
    at IOSession.Session.onMessage (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154988:35)
    at Interface.<anonymous> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:157204:27)
    at Interface.emit (events.js:223:5)
    at Interface._onLine (readline.js:315:10)
    at Interface._normalWrite (readline.js:460:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:223:5)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:290:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:181:23)
	at Function.create (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:76476)
	at dispatchResponse (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:70351)
	at dispatchMessage (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:69210)
	at /usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:68720
	at l.fire (/usr/share/code/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:46:67)
	at g.onLengthData (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:45:60018)
	at Socket.<anonymous> (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:45:59666)
	at Socket.emit (events.js:223:5)
	at addChunk (_stream_readable.js:309:12)
	at readableAddChunk (_stream_readable.js:290:11)
	at Socket.Readable.push (_stream_readable.js:224:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:181:23)
[2020-12-07 16:59:29.513] [renderer6] [error] <syntax> TypeScript Server Error (4.0.3)
Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
Error: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
    at addSyntheticNodes (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142388:30)
    at createChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142377:9)
    at NodeObject.getChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142321:56)
    at Object.findChildOfKind (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:111269:26)
    at spanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123467:36)
    at getOutliningSpanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123387:28)
    at visitNonImportNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123233:28)
    at addNodeOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123204:21)
    at Object.collectElements (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123191:13)
    at Proxy.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:143739:50)
    at IOSession.Session.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154172:45)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:153172:61)
    at /usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:88
    at IOSession.Session.executeWithRequestId (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154955:28)
    at IOSession.Session.executeCommand (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:33)
    at IOSession.Session.onMessage (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154988:35)
    at Interface.<anonymous> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:157204:27)
    at Interface.emit (events.js:223:5)
    at Interface._onLine (readline.js:315:10)
    at Interface._normalWrite (readline.js:460:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:223:5)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:290:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:181:23): Error: <syntax> TypeScript Server Error (4.0.3)
Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
Error: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
    at addSyntheticNodes (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142388:30)
    at createChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142377:9)
    at NodeObject.getChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142321:56)
    at Object.findChildOfKind (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:111269:26)
    at spanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123467:36)
    at getOutliningSpanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123387:28)
    at visitNonImportNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123233:28)
    at addNodeOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123204:21)
    at Object.collectElements (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123191:13)
    at Proxy.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:143739:50)
    at IOSession.Session.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154172:45)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:153172:61)
    at /usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:88
    at IOSession.Session.executeWithRequestId (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154955:28)
    at IOSession.Session.executeCommand (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:33)
    at IOSession.Session.onMessage (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154988:35)
    at Interface.<anonymous> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:157204:27)
    at Interface.emit (events.js:223:5)
    at Interface._onLine (readline.js:315:10)
    at Interface._normalWrite (readline.js:460:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:223:5)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:290:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:181:23)
	at Function.create (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:76476)
	at dispatchResponse (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:70351)
	at dispatchMessage (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:69210)
	at /usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:68720
	at l.fire (/usr/share/code/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:46:67)
	at g.onLengthData (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:45:60018)
	at Socket.<anonymous> (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:45:59666)
	at Socket.emit (events.js:223:5)
	at addChunk (_stream_readable.js:309:12)
	at readableAddChunk (_stream_readable.js:290:11)
	at Socket.Readable.push (_stream_readable.js:224:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:181:23)
[2020-12-07 17:01:43.325] [renderer6] [error] <syntax> TypeScript Server Error (4.0.3)
Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
Error: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
    at addSyntheticNodes (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142388:30)
    at createChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142377:9)
    at NodeObject.getChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142321:56)
    at Object.findChildOfKind (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:111269:26)
    at spanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123467:36)
    at getOutliningSpanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123387:28)
    at visitNonImportNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123233:28)
    at addNodeOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123204:21)
    at Object.collectElements (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123191:13)
    at Proxy.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:143739:50)
    at IOSession.Session.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154172:45)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:153172:61)
    at /usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:88
    at IOSession.Session.executeWithRequestId (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154955:28)
    at IOSession.Session.executeCommand (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:33)
    at IOSession.Session.onMessage (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154988:35)
    at Interface.<anonymous> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:157204:27)
    at Interface.emit (events.js:223:5)
    at Interface._onLine (readline.js:315:10)
    at Interface._normalWrite (readline.js:460:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:223:5)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:290:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:181:23): Error: <syntax> TypeScript Server Error (4.0.3)
Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
Error: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
    at addSyntheticNodes (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142388:30)
    at createChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142377:9)
    at NodeObject.getChildren (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:142321:56)
    at Object.findChildOfKind (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:111269:26)
    at spanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123467:36)
    at getOutliningSpanForNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123387:28)
    at visitNonImportNode (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123233:28)
    at addNodeOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123204:21)
    at Object.collectElements (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:123191:13)
    at Proxy.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:143739:50)
    at IOSession.Session.getOutliningSpans (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154172:45)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:153172:61)
    at /usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:88
    at IOSession.Session.executeWithRequestId (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154955:28)
    at IOSession.Session.executeCommand (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154964:33)
    at IOSession.Session.onMessage (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:154988:35)
    at Interface.<anonymous> (/usr/share/code/resources/app/extensions/node_modules/typescript/lib/tsserver.js:157204:27)
    at Interface.emit (events.js:223:5)
    at Interface._onLine (readline.js:315:10)
    at Interface._normalWrite (readline.js:460:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:223:5)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:290:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:181:23)
	at Function.create (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:76476)
	at dispatchResponse (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:70351)
	at dispatchMessage (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:69210)
	at /usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:1:68720
	at l.fire (/usr/share/code/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:46:67)
	at g.onLengthData (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:45:60018)
	at Socket.<anonymous> (/usr/share/code/resources/app/extensions/typescript-language-features/dist/extension.js:45:59666)
	at Socket.emit (events.js:223:5)
	at addChunk (_stream_readable.js:309:12)
	at readableAddChunk (_stream_readable.js:290:11)
	at Socket.Readable.push (_stream_readable.js:224:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:181:23)
```

I kept watching when they appear
1. First one happens when I open file
2. Second one has appeared when I notice fold/unfold has gone
3. Third one appears when I hit the save

</p></details>

Update: I tried get some [typescript logs](https://github.com/microsoft/vscode/files/5656218/logs.2.zip)

**Comment by 5saviahv** [comment]
I tried to understand what the error means so I took some class and made it simpler and simpler. Until I realised It seems to happen if I declare anything through prototype.

I though this error is probably not related until same thing happened with this simple class below.
```
class Vaata {
    constructor(name = "") {
        this.name = name;
    }
}

/* Set a some simple signature for the object */
Vaata.prototype.vaata = '[object Vaata]';
```

**Comment by 5saviahv** [comment]
I also updated typescript to 4.1.2 from npm and same happens.

Similar issues wont give lot:

[TS format doc error (#102506)](https://github.com/microsoft/vscode/issues/102506)
[Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia](https://github.com/microsoft/TypeScript/issues/35133)
[Error: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia ...](https://github.com/microsoft/TypeScript/issues/35516)

**Comment by 5saviahv** [comment]
Error: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia

This error affects UI some way, but typescript 3.6.4, seems to cured it.

**Comment by aeschli** [comment]
I'm able to reproduce the TypeScript error, but not the folding issue.
- VSCode stable 1.51.1 (TypeScript 4.0.3)
- download and extract https://nodejs.org/dist/v10.15.0/node-v10.15.0.tar.gz, open VSCode on it
- open `lib/_stream_readable.js`
- fold all elements (Ctrl + k, Ctrl + 0)
- put cursor on function ReadableState *line 69)
- Invoke code action `Convert all constructor functions to classes`, save file
- reload vscode and open the output view

I don't see any stale folding regions

**Comment by 5saviahv** [comment]
This folding issue happens 15..30 minutes, after you open document. I just kept window on my side monitor as information source and all the sudden folding ability is lost and if you look "log (window)" this error has repeated itself.

It seems like vscode updates its memory, but this time error causes vscode looses data where functions start or end.

I also tried get some "verbose log" and added it on the end of my first post. I am no expert but it was requested with similar issues so I added it. interesting note with typescript 3.6.4 it has been over 17 hours: no error and everything still works.
